File: code-samples/voice/code-snippets/call-monitoring-group.cs
using System;
using System.IO;
using System.Threading.Tasks;
using System.Collections.Generic;
using RingCentral;
using Newtonsoft.Json;

namespace CallMonitoringGroup {
  class Program {
    static RestClient restClient;
    static async Task Main(string[] args)
    {
      try
      {
        // Instantiate the SDK
        restClient = new RestClient("SANDBOX-APP-CLIENT-ID", "SANDBOX-APP-CLIENT-SECRET", "https://platform.devtest.ringcentral.com");

        // Authenticate a user using a personal JWT token
        await restClient.Authorize("SANDBOX-JWT");

        await read_call_monitoring_groups();
      }
      catch (Exception ex)
      {
        Console.WriteLine("Unable to authenticate to platform. Check credentials. " + ex.Message);
      }
    }
    /*
    * Read all call monitoring groups
    */
    static private async Task read_call_monitoring_groups()
    {
      try
      {
        var resp = await restClient.Restapi().Account().CallMonitoringGroups().Get();
        foreach (var group in resp.records)
        {
          Console.WriteLine($"Call monitoring group name/id: {group.name}/{group.id}");
          await read_call_monitoring_group_members(group.id);
        }
      }
      catch (Exception ex)
      {
        Console.WriteLine($"Unable to list call monitoring groups. {ex.Message}");
      }
    }
    /*
    * Read a call monitoring group members
    */
    static private async Task read_call_monitoring_group_members(String groupId)
    {
      try
      {
        var resp = await restClient.Restapi().Account().CallMonitoringGroups(groupId).Members().Get();
        Console.WriteLine("Call monitoring group members:");
        foreach (var member in resp.records)
        {
          Console.WriteLine(JsonConvert.SerializeObject(member));
        }
      }
      catch (Exception ex)
      {
        Console.WriteLine($"Unable to read members of this call monitoring group. {ex.Message}");
      }
    }
  }
}

File: code-samples/voice/code-snippets/call-monitoring-group.java
package CallMonitoringGroup;

import java.io.IOException;
import com.google.common.reflect.TypeToken;
import com.google.gson.Gson;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class CallMonitoringGroup {
  static RestClient restClient;

  public static void main(String[] args) {
    var obj = new CallMonitoringGroup();
    try {
      // Instantiate the SDK
      restClient = new RestClient("SANDBOX-APP-CLIENT-ID", "SANDBOX-APP-CLIENT-SECRET", "https://platform.devtest.ringcentral.com");

      // Authenticate a user using a personal JWT token
      restClient.authorize("SANDBOX-JWT");

      obj.read_call_monitoring_groups();

    } catch (RestException e) {
      System.out.println(e.getMessage());
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
  /*
  * Read all call monitoring groups
  */
  public void read_call_monitoring_groups() throws RestException, IOException {
    try {
      var resp = restClient.restapi().account().callMonitoringGroups().get();
      for (var group : resp.records)
      {
        System.out.println("Call monitoring group name/id: " + group.name + "/" + group.id);
        read_call_monitoring_group_members(group.id);
      }
    } catch (RestException ex) {
      System.out.println("Unable to list call monitoring groups. " + ex.getMessage());
    }
  }
  /*
  * Read a call monitoring group members
  */
  public void read_call_monitoring_group_members(String groupId) throws RestException, IOException {
    try {
      var resp = restClient.restapi().account().callMonitoringGroups(groupId).members().get();
      System.out.println("Call monitoring group members:");
      for (var member : resp.records)
      {
        String jsonStr = new Gson().toJson(member, new TypeToken<Object>(){}.getType());
        System.out.println(jsonStr);
      }

    } catch (RestException ex) {
      System.out.println("Unable to read members of this call monitoring group. " + ex.getMessage());
    }
  }
}
